clang/test/CodeGenSYCL/esimd-accessor-ptr-md.cpp

```diff
@@ -0,0 +1,30 @@
+// RUN: %clang_cc1 -fsycl -fsycl-explicit-simd -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -emit-llvm %s -o - | FileCheck %s
+
+// This test checks proper metadata generation for accessors used in ESIMD
+// kernels:
+// 1) Proper 'kernel_arg_accessor_ptr' metadata is generated by the FE for ESIMD
+//    kernels
+// 2) Pointers originating from accessors are marked with 'buffer_t' and proper
+//    argument kind.
+
+#include "Inputs/sycl.hpp"
+
+void test(int val) {
+  cl::sycl::accessor<int, 1, cl::sycl::access::mode::read_write> accessorA;
+  cl::sycl::accessor<int, 1, cl::sycl::access::mode::read> accessorB;
+
+  cl::sycl::kernel_single_task<class esimd_kernel>(
+      [=]() __attribute__((sycl_explicit_simd)) {
+        accessorA.use(val);
```

Comment:
Could you please also try other kinds of arguments (like POD for example) and check that all metadata is correct.

Reply:
Is this a nit? The main point of the test is checking accessors argument translation and MD generation. PODs are no different from other types in this regard - and `int val` should be enough.

Reply:
My point was to check that we don't generate something unnecessary for non-accessor types in case if it can confuse underlying backend. And I probably missed that you already have captured int, so yes, this comment is more a nit.